cpp/ql/src/semmle/code/cpp/dataflow/internal/DataFlowImplCommon.qll

```diff
@@ -512,13 +512,19 @@ abstract class CallContext extends TCallContext {
   abstract predicate relevantFor(DataFlowCallable callable);
 }
 
-class CallContextAny extends CallContext, TAnyCallContext {
+abstract class CallContextNoCall extends CallContext { }
```

Comment:
Couldn't we define this simply with a charpred of `not this instanceof CallContextCall` instead of an abstract class? I think that would be slightly clearer.

Reply:
`CallContext` is itself `abstract`, so that doesn't work nicely.